Enable CORS headers http referer replace and middleware

melvinsoft opened the pull request:
This PR includes settings to allow us to use CORS headers requests when we're running under HTTPS. When we're using django-cors-headers properly configured, we're able to retrieve the csrf external cookie, but when we want to make a POST request from the external domain is working under HTTP but not under HTTPS since the referer is passed, and is blocked if it's external. 
See the [Django's docs](https://docs.djangoproject.com/en/1.11/ref/csrf/) (point 4)

In Django 1.9 a [new settings](https://docs.djangoproject.com/en/1.11/ref/settings/#std:setting-CSRF_TRUSTED_ORIGINS) has been added to allow to set the white listed domains that can make request, but since this settings is not available until Django 1.9, we need to use the [django-cors-headers workaround](https://github.com/ottoyiu/django-cors-headers#cors_replace_https_referer).

melvinsoft commented:
![](https://github.trello.services/images/mini-trello-icon.png) [(2) PSU: Final steps to support site launch](https://trello.com/c/OYKDkfgg/1554-2-psu-final-steps-to-support-site-launch)

tkeemon commented on lms/envs/aws_appsembler.py:
```diff
@@ -104,3 +104,14 @@
     # removing the database alias
     del DATABASES['appsembler_usage']
+
+if FEATURES.get('ENABLE_CORS_HEADERS'):
```
Let's set a default value for this get method in case it's not correctly set in the env:

```
FEATURES.get('ENABLE_CORS_HEADERS', False)
```

melvinsoft commented:
@tkeemon Thanks for reviewing this, I've added the default for the settings, and also I've removed the middleware, I know is recommended, but not mandatory, and breaking the external requests again, and working without it, so I think we can remove it.
